include/circt/Dialect/FIRRTL/FIRRTLTypes.h

```diff
@@ -292,11 +292,14 @@ class BundleType : public FIRRTLType::TypeBase<BundleType, FIRRTLType,
   /// Look up an element's index by name.  This returns None on failure.
   llvm::Optional<unsigned> getElementIndex(StringRef name);
 
+  /// Look up an element's name by index. This returns None on failure.
+  llvm::Optional<StringRef> getElementName(size_t index);
```

Comment:
The only client of this is force unwrapping the optional.  I'd recommend making this assert that the index is valid, and returning a StringRef directly.